lib/test-utils/src/Test/QuickCheck/Extra.hs

```diff
@@ -249,6 +252,27 @@ partitionList (x, y) =
     x' = max 0 x
     y' = max 1 (max y x')
 
+--------------------------------------------------------------------------------
+-- Selecting random map entries
+--------------------------------------------------------------------------------
+
+-- | Selects an entry at random from the given map.
+--
+-- Returns the selected entry and the remaining map with the entry removed.
+--
+-- Returns 'Nothing' if (and only if) the given map is empty.
+--
+mapEntry :: forall k v. Ord k => Map k v -> Gen (Maybe ((k, v), Map k v))
```

Comment:
I think this family of functions should be called `getMapEntry` or similar. `mapEntry` is a bit similar to `map` (i.e `fmap`), which has the connotation of "mapping over" something. I think if it was named `fmapEntry` this erroneous connotation would be more obvious.

Just an opinion, not a blocker.

Reply:
Thanks @sevanspowell, this is great feedback.

I've changed the names in 44df8f839404e19c3223600d525f00f5a1d21015.